### ISSUE TITLE
dayplans: blank screen or blank image container after adding an image

### ISSUE THREAD

**Issue by urapadmin** [urapadmin/kiosk#1246]
this comes out of #1151 which got out of hand.

This is where we are: 
We have two ipads Luiza's and Laurel's which were able to reproduce the one or the other issue reliably until I release
test-2 which used a new way of processing the file-identifier-cache during image creation. Instead of opening a flickering overlay window I changed the layout in situ. The side effect was a short but in fact unavoidable tab switch. 
While this is a bad way of doing it for me it worked reliably on Luiza's ipad but Laurel got a blank screen once (out of several tries).

Pretty devastating.

Now this here is about a third take:
### recording 11.5-test-3
- [ ] in the Guinea pig section you still find "dayplan_in_situ", which, if checked solely, still does what 11.5.-test-2 did. That's the mode that failed once on Laurel's ipad. 👉 In this mode the excavating form shows "experimental" in the upper right corner.
- [ ] if you check dayplan_deferred (Please only check one of the options) the processing is not done in situ but with a delay that is supposed to give filemaker time to "settle" (we are back to the time when computing was a rather analog thing you should not do too speedy). You should see one slightly delayed flicker. After that your image should be fine and with the correct identifier in the gallery. 👉 In this mode the excavating form shows "**totally** experimental" in the upper right corner.

**Comment by lbestock** [comment]
I was able to break this again (while sheets dry and the small one sleeps on the sofa). HOWEVER, I think you may be onto it now and so I will describe my steps with a bit of optimism for the first time.
I downloaded the new file.
I accidentally made a dayplan before going to Guinea Pig and it worked.
Please note that nearly every app was closed right now except for FM Go.
I tried with both dayplan_in_situ and then with dayplan_deferred checked and made seven new dayplans in each case. All worked perfectly.
I left the FM file open but went and opened a ton of other apps, including one that ran a video in a mini window so I could make sure I was stressing it a lot.
Both failed reliably. I got blank image containers and a total list of blanks in dayplan_deferred, and the same thing in dayplan_in_situ. 
I stopped the video and tried again, but still got the same kind of failure in both places. 
I then closed a bunch of apps again, all without closing the FM file or anything, and then went and tried again and had success. 
So for the first time we seem to have a variable that we can use to provoke this behavior, and it is having too much other stuff running. The amount dayplan_deferred slowed things down was insufficient to deal with the stress I put it under with many apps running. I did not observe a difference between dayplan_deferred and dayplan_in_situ, but I have not done microscopic stress testing just this gross one that suggests we may have a bit of light now.

**Comment by luizaogs** [comment]
Goddammit. I was trying to check whether having different applications running would lead to the same results on my iPad and my first dayplan under dayplan_in_situ was a blank container, a problem I did not have at all yesterday after adding 10 dayplans. I only have Safari and Files open other than FM, which should really not be that much stress? Let's see what happens when I close them...nope, still blank. What the hell. So now I've had two blank containers in this mode whereas yesterday I had none, and I cannot figure out why. Logged out, logged in, still blank containers for the old images as well as the one I added now.

Jesus Christ. I still only have FM open, but not even dayplan_deferred is working. Blank container. What the hell.

Ok. Turned my iPad OFF, turned it back on. First try after that in dayplan_in_situ worked as it should. Second one did not, however -- blank container. I still have only FM open. Ok just kidding, I realized I had the Apple Store open. Once I closed that app and added a third dayplan, it worked fine. Five more worked normally after that. Now I'm going to open a bunch of apps (Netflix, TouchDraw, Photos, FaceTime, GoogleMeet. Oof). Hmm, that still worked for me. Not enough stress? I have no idea. So then I also opened AppleTV, Apple Store, Books, Facebook (ew), Youtube. Ok, now I got an empty container. But just one, and the next dayplan I added worked. 

Dayplan_deferred time, still with all of that crap open. First dayplan resulted in a blank container, but my second one worked. So I am not sure if the stress works as a trigger, at least on my iPad. Next two dayplans were also blank containers, but the next one was not. So then I closed my million apps but did not log out of FM and back in, and my first dayplan there was a blank container. But then the second one worked? Then another one worked, then the next one did not. Then the next five did. I really do not understand. Logged out of FM, added one that worked, but the next one did not. 

Turned my iPad off again. Turned on dayplan_deferred. Added six dayplans, all of which worked. Turned on dayplan_in_situ. Added six more dayplans, all of which also worked. Logged out of FM, logged back in, and was able to reliably add some more dayplans in both Guinea Pig categories.

So my conclusion here is that turning the iPad off and on seems to help, though note that above that did not resolve the issue reliably either when I had another app running at the same time.

**Comment by urapadmin** [comment]
from my perspective it is just another round in the mess. We never get to a reliable factor that actually solves it. It seems like getting better at times. But this is the digital world and in that world that does not mean anything. So as long as we have a single negative result we have a potential problem that can escalate in future.

That being said, we need to distinguish the errors better:
1. blank screen - that's the worst. FM is covered by white with things still happening underneath the white sheet (we know that from the menu that pops up).
2. blank container. Here the image does not __immediately__ appear in the __container it was inserted into__. This comes in two variations:
  2a. temporary: the image is there after leaving the form and going back. So the image has been inserted but isn't displayed at times.
  2b. permanent: the image isn't there, even after leaving filemaker (something that so far only Luiza seems to be encountering). So it seems the image really has not even been inserted (the most mysterious of all of them). 
3. blank list: the other dayplans __in the list__ turn to white - that only happens temporarily, so when leaving FM and going back they are there (so we know they have been inserted just sometimes they are not displayed).

Is that right so far? Then please in future refer to those types. I have a hunch that we might even have different issues playing into each other making the whole thin foggier and foggier.

**Comment by luizaogs** [comment]
I think that is right, yes.

**Comment by luizaogs** [comment]
Oh though I will say, for the permanent blank containers (2b), that I get them in the gallery - they’re all just blank. So it seems like something has indeed been added? I don’t know.

**Comment by urapadmin** [comment]
that is interesting, indeed. Next time you get 2b please stop and send that fmp12 file to me (upload to file repository makes some sense for I am not sure if github allows such woppers).

**Comment by urapadmin** [comment]
### recording 11.5-test-4
This new version makes the "deferred" (aka totally experimental) mode the default strategy. If you want the original 11.5 mode to see if the issue is still there you switch on "standard 11.5" under guinea pigs.

This version starts with a new log whenever you insert a new dayplan image. Otherwise the logs get unreadable for me. Should you encounter one of our errors, press the log button afterwards (unfortunately this does not work with error type 1) and copy the log. It should be rather short.

Please make a fresh round of tests:

- [ ] with "deferred less code" switched off
- [ ] with "deferred less code" switched on

"deferred less code" will not add the identifier to the image, so you won't find the new image under the identifier in the gallery. I am still trying to find out if my code has anything to do at all with the issue. If we get one of the error types above only once with "deferred less code" switched on, then we narrow it down more and more to a problem that simply occurs when inserting an image in filemaker. That would be the worst case.

**Comment by urapadmin** [comment]
Hopefully I'll get my own 2019 mini on Wednesday so - more hope - we don't have to try every single thing across the ocean any longer.

**Comment by luizaogs** [comment]
Ok, with deferred less code off: first try was a blank container (so I'll be giving you this database shortly, since I have a blank container in the gallery). The next one in that mode was also a blank container. Logged out and came back and the images were not there, so both are of the type 2b.

Deferred less code on: first dayplan added was also a blank container (2b). Logged out and came back in, another 2b.

So in fact I've now added five dayplans across both modes and they have all been 2bs.

Lutz, database coming to you. I've been working in FL (Playground) so that's how you'll find the blank images in the gallery.

**Comment by urapadmin** [comment]
Wow, you are so fast. Perhaps having my own mini won't make things faster. 
Can you please check the memory usage of your ipad and add a screen shot of the settings of "camera" here?

**Comment by urapadmin** [comment]
oh, and please paste a log here of your 2b, Luiza.

**Comment by luizaogs** [comment]
That's because I don't sleep :) Is this what you meant? I'm somewhat stupid.

![IMG_0223](https://user-images.githubusercontent.com/67970402/140645815-9a1ad4f3-cd17-44c5-a589-5933149711c8.PNG)

![IMG_0224](https://user-images.githubusercontent.com/67970402/140645816-b8c518a8-15b7-49bf-9f40-a5255b33dd31.PNG)

Also, you'll find the database in the file repository under "nightmares" because I think I had one about disappearing images.

**Comment by urapadmin** [comment]
no, you are not. That is exactly what I wanted. Only I always want more: what is behind the "Formats" in Camera? I don't have that one on my ipad.

**Comment by urapadmin** [comment]
@lbestock: I need those two screenshots from you, too.

**Comment by luizaogs** [comment]
Ah yes, sorry! I meant to ask about the log and forgot because I did not see the button. Should I be in test mode?

**Comment by urapadmin** [comment]
yes, I think you need to be in test mode.

**Comment by luizaogs** [comment]
![IMG_EA57697D85A1-1](https://user-images.githubusercontent.com/67970402/140645924-465406de-4236-4c88-bb70-8a2ea833ca0f.jpeg)

**Comment by urapadmin** [comment]
ah. We still have the HEIC thingy going on. Old "friend" of mine.
Well, after may run a round of tests with "Most Compatible" just out of curiosity.

**Comment by luizaogs** [comment]
Ok, here is the short log for a 2b with deferred less code on:

add_dayplan_to_cache_deferred: quitting early.
image_validate: field img
add_dayplan_to_cache_deferred: quitting early.

**Comment by lbestock** [comment]
![68A57CAA-FF87-4D0D-A974-15F176F33B2E](https://user-images.githubusercontent.com/49291909/140649280-8c1ea61f-1929-4e45-8bb2-9944025f95cd.png)
![768B7412-04B5-4305-B69F-C2E8810910A7](https://user-images.githubusercontent.com/49291909/140649281-4d25e7db-2f97-4529-81b7-926dfa368f68.png)
And I appear to be shooting HEICs, too.

**Comment by urapadmin** [comment]
Laurel, after you ran the tests above and encounter 1 or 2 can you please do a separate run with "Keep Normal Photo"?
And after that you could do a third one switching off Smart HDR. Something nobody has but you.

**Comment by urapadmin** [comment]
reminds me, how many databases do you have in filemaker? You should try to have only the one you are testing with at all stored in filemaker itself.

**Comment by urapadmin** [comment]
### 11.5.test-4 blank in file repository
The same version that is used by the field manager is now available as a blank database in the file repository under the usual tag name. Try that one, please. And if it yields the same or similar results we will focus on the blank one in future.

**Comment by luizaogs** [comment]
In blank database: 

I had neither the standard 11.5 nor the deferred less code checked, and I've been able to add two dayplans normally. The third one was a 2b.

With deferred less code on, I got a 2b immediately.

**Comment by urapadmin** [comment]
have you given me a database after a 2b? Can't find one in the file repository.

You never get anything but a 2b it seems.

**Comment by luizaogs** [comment]
You didn’t find one under the “nightmares” tag?

**Comment by urapadmin** [comment]
I overlooked the message about the tag and was looking with "latest first" sort order. That really does not lead to useful results on pvd. But that's another story...

**Comment by urapadmin** [comment]
can't remember, have you seen the error type 1 at all, ever?

**Comment by luizaogs** [comment]
Yeah, a lot, but not lately. I documented some of that with different keyboards in the previous ticket about this, I think.

**Comment by urapadmin** [comment]
okay, the 2b dayplans are actually not there. No image has been transferred from the camera to filemaker. 
When you have time please try switching off "High efficiency" in your camera options and see if that has any effect on your 2b production.

**Comment by urapadmin** [comment]
> 
> 
> Yeah, a lot, but not lately. I documented some of that with different keyboards in the previous ticket about this, I think.

hm. Please try to use "standard 11.5" to see if that still happens. I think you had the best luck with the floating keyboard? Or was that about 2b. Hmpf. 

You can use the blank database.

**Comment by luizaogs** [comment]
> okay, the 2b dayplans are actually not there. No image has been transferred from the camera to filemaker. When you have time please try switching off "High efficiency" in your camera options and see if that has any effect on your 2b production.

No, or at least apparently not. Two dayplans added with "high efficiency" turned off and "most compatible" turned on were 2bs.

**Comment by luizaogs** [comment]
> > Yeah, a lot, but not lately. I documented some of that with different keyboards in the previous ticket about this, I think.
> 
> hm. Please try to use "standard 11.5" to see if that still happens. I think you had the best luck with the floating keyboard? Or was that about 2b. Hmpf.
> 
> You can use the blank database.

Yep, got the blank screen (error 1) on my first try with "standard 11.5" on. Regular keyboard. In your description of the error above you say that we get the menu, but I think I've been getting only the keyboard popping up most frequently. Once I quit FM the image was indeed there, however.

**Comment by urapadmin** [comment]
### Quick summary:
- okay. So far I have the feeling that error type 1 is by and large gone with the deferred strategy. We have only one encounter on Laurel's ipad.
- I also have the feeling that 2b is only present on Luiza's ipad. 
- HEIC / not HEIC has no effect, so that is off the table

**Comment by lbestock** [comment]
I have tested 11.5-test-4 for 20 minutes without being able to produce any errors, except when I had clicked "standard 11.5" under Guinea Pigs (then I got an error 1, which was temporary in that I logged back in and found my image there. I had the blank with keyboard, not blank with menu, variety.) I tested deferred less code off and on, I tested both with and without lots of other apps running, and all my dayplans worked perfectly.

I confirm that 2b is an error type I have not ever seen.

**Comment by lbestock** [comment]
Oh my god I cannot provoke any of this today. I haven't changed my camera settings or anything because nothing I do makes the error come. Blank database or workstation. DLC on or off. Lots of apps running or none. Today it does what it is supposed to and having added approximately 20 pictures of Omar in unfortunate positions I want to tear my own hair out as thoroughly as she does hers.

**Comment by urapadmin** [comment]
that confirms at least my feeling that 1 is effectively battled by doing the file-identifier-cache updates asynchronously so that filemaker can breathe. It is just too much too quickly for the old rattletrap. 

I tend to make different tickets for 1 and 2b. 

I wonder, does 2a exist at all? Have you guys ever had it?

**Comment by urapadmin** [comment]
It is all really a bit like making cappuccino only in case of filemaker it is really not worth the effort ever.

**Comment by lbestock** [comment]
Oh yes, 2a exists and for a long time was the most common version for me.

**Comment by luizaogs** [comment]
Yes, I think I've had it as well but now really only get 2b.

**Comment by urapadmin** [comment]
> 
> 
> Oh yes, 2a exists and for a long time was the most common version for me.

but you never got 2b. So I have a hunch that 2a is related to 1 while 2b is related to, erm..., Luiza :o) Just kidding. But 2b seems particularly weird. It is a pity that you two have not had the change to test next to each other to perhaps find a difference in using it.

**Comment by lbestock** [comment]
Simultaneous testing in one place of three iPads using identical steps resulted in three different results.
Cleared everything out on the iPads.
Downloaded the blank DB horrors.
Entered and checked nothing, just went straight into PG (so no checking of deferred less code).
Made a new image container.
Clicked the container and then camera.
Took really nearly the identical photo.
Purple ipad: worked totally fine.
Luiza's ipad: has a some sort of 2 that doesn't look like it did anything, what she has called 2b (but we're not leaving yet so she can't check that it hasn't come back yet).
My iPad: also looks like a 2 of some sort but is different than Luiza's in two respects. One, the image in the list is white, not grey. Two, the pin is there and the "insert image" text gone from the image container, while in Luiza's the "insert image" text is not there at all - it looks like nothing happened.

**Comment by lbestock** [comment]
Now we've gone out and back in again, without logging out or anything.
Purple everything looks just fine.
Luiza has an empty image container in both the dayplans and the gallery.
And mine everything looks normal.

Now deferred less code, coming up.

**Comment by lbestock** [comment]
Now with DLC checked after a fresh download so that we were definitely working in the same file, not the one that had been differently altered above. Now we get two versions on three iPads.
Laurel and purple: fine, as it should be.
Luiza got a 2b again.

**Comment by lbestock** [comment]
But then for extra special fun, on my iPad I clicked the plus button to make a new image container. And before I even clicked in it or had the chance to go to camera I got what looks like a 3 - blank in the list and a white image container with no "insert image here" text. I hadn't even clicked the image container!

**Comment by urapadmin** [comment]
with the last one I would ask you to just switch to a different tab and back.

**Comment by lbestock** [comment]
Hmpf we were already down to the next wipeout and try again with a fresh download so I can't. And I can't repeat it naturally.
I forgot to report stuff:
Purple is iOS 15.1
Luiza is iOS 15.0.2
Laurel is iOS 14.8

So we're in for a round of updating iOSs next.

**Comment by lbestock** [comment]
New test with all three iPads in iOS 15.1
Fresh downloads.
Camera settings were changed before we tried any of this. The three iPads can't be made identical but we did our best. We made format most compatible. We turned live photo off where possible (not an option for purple). We made keep normal photo on for all of them.
First round we had DLC checked. We made three photos and a signature each. Everything worked as it should.
Second round (we did not redownload or exit, since our databases were still comparable) we unchecked DLC. Two more dayplans made, all three ipads behaved the same way and it was proper.
We then changed the Guinea Pig rule to check Standard 11.5.
This was the first divergence we saw. Purple worked properly, and Luiza and Laurel did not - we had a #1 error with the keyboard and blank screen. In both cases the image was in fact there when we left the tab and went back in.

**Comment by luizaogs** [comment]
I was worried that mine would stop working and result in 2bs again (because it often works properly right after the iPad is turned off and on and then doesn't), but so far I've been able to add several perfectly fine dayplans both with nothing checked and with DLC on after leaving Laurel's. I'll can try again tomorrow...

**Comment by luizaogs** [comment]
Oh, we also made the camera setting "full" -- just so I don't forget, in case I need to reconstruct a combination that worked.

**Comment by urapadmin** [comment]
![9C3A0810-5531-471D-8868-B0596559DE7C](https://user-images.githubusercontent.com/38838314/140938213-517e7c6a-bb77-4dfe-9a3c-8bced563d8ff.png)

**Comment by urapadmin** [comment]
Just for the records, what we want is that it works with nothing checked. DLC is not adding anything to Filemaker

**Comment by urapadmin** [comment]
Fun fact: My old mini 4 that never shows any of the error types is on Luiza's problematic 15.0.2 version that seems to have caused all the 2bs. It is such a mess.

**Comment by urapadmin** [comment]
okay, I'd say let's split this up now into 2b -> which we will consider solved by a new OS version.
and 1 -> which is still there and needs an overhaul of all image containers.
For the latter I create a new ticket, for the former we use this here and I close it.

**Comment by urapadmin** [comment]
new ticket for error type 1: #1249.

I close this and we go with updating the OS to something else than 15.0.2 for type 2b. 

2a we have not seen anymore and 3 I ignore for the time being.
